test/test_sparse.py

```diff
@@ -823,6 +823,69 @@ def test_shape(sparse_dims, nnz, with_size):
         test_shape(4, 10, [100, 100, 100, 5, 5, 5, 0])
         test_shape(4, 0, [0, 0, 100, 5, 5, 5, 0])
 
+    @skipIfRocm
+    def test_sparse_sum(self):
+        nnz = 10
+        sparse_dims = 2
+        with_size = [5, 5, 3, 4]
+        S, _, _ = self._gen_sparse(sparse_dims, nnz, with_size)
+        S = S.coalesce()
```

Comment:
Every test in this class is run twice: once with `_gen_sparse` generating coalesced values, and once with `_gen_sparse` generating uncoalesced values. So if you only want to run when things are coalesced, it would be better to test for `self.is_uncoalesced` to avoid uselessly running this test a second time. We could probably turn this into a decorator, if you were so inclined.

Reply:
Additionally, your implementation of sparse sum on all dimensions seems to work on the uncoalesced case too, so it would be good to test it in that case.

Reply:
Thanks for both valuable inputs! Will change accordingly